node-ui/src/hooks/useInternetComputer.tsx

```diff
@@ -0,0 +1,352 @@
+const possibleNetworks = {
+  production: {
+    url: 'https://identity.ic0.app',
+    cannisterId: 'rdmx6-jaaaa-aaaaa-aaadq-cai',
+  },
+  staging: {
+    url: 'https://beta.identity.ic0.app/',
+    cannisterId: 'fgte5-ciaaa-aaaad-aaatq-cai',
+  },
+};
+
+export type NetworkId = keyof typeof possibleNetworks;
+
+export function useInternetComputer(): useInternetComputerReturn {
+  const navigate = useNavigate();
+  const [ready, setReady] = useState<boolean>(true);
+  const { showServerDownPopup } = useServerDown();
+  const [walletSignatureData, setWalletSignatureData] =
+    useState<WalletSignatureData | null>(null);
+  const [signData, setSignData] = useState<SignData | null>(null);
+
+  const [currentNetwork, setCurrentNetwork] = useState(
+    possibleNetworks['production'],
+  );
+
+  const requestNodeData = useCallback(
+    async ({ setErrorMessage }: RequestNodeDataProps) => {
+      const challengeResponseData: ResponseData<NodeChallenge> =
+        await apiClient(showServerDownPopup).node().requestChallenge();
+      const { publicKey } = await getOrCreateKeypair();
+
+      if (challengeResponseData.error) {
+        console.error(
+          `${t.requestNodeDataError}: ${challengeResponseData.error}`,
+        );
+        setErrorMessage(
+          `${t.requestNodeDataError}: ${challengeResponseData.error}`,
+        );
+        return;
+      }
+
+      const signatureMessage: SignatureMessage = {
+        nodeSignature: challengeResponseData.data?.nodeSignature ?? '',
+        publicKey: publicKey,
+      };
+
+      const signatureMessageMetadata: SignatureMessageMetadata = {
+        publicKey: publicKey,
+        nodeSignature: challengeResponseData.data?.nodeSignature ?? '',
+        nonce:
+          challengeResponseData.data?.nonce ?? randomBytes(32).toString('hex'),
+        timestamp:
+          challengeResponseData.data?.timestamp ?? new Date().getTime(),
+        message: JSON.stringify(signatureMessage),
+      };
+      const signatureMetadata: InternetComputerSignatureMessageMetadata = {};
+      const payload: Payload = {
+        message: signatureMessageMetadata,
+        metadata: signatureMetadata,
+      };
+      const wsd: WalletSignatureData = {
+        payload,
+        publicKey,
+      };
+      setWalletSignatureData(wsd);
+    },
+    [showServerDownPopup],
+  );
+
+  const changeNetwork = useCallback(
+    async (networkId: NetworkId, setErrorMessage: (msg: string) => void) => {
+      try {
+        setReady(false);
+        setErrorMessage('');
+        console.log('Changing network to:', networkId);
+        setCurrentNetwork(possibleNetworks[networkId]);
+      } catch (error) {
+        console.error(`${t.errorChangingNetwork}: ${error}`);
+        setErrorMessage(`${t.errorChangingNetwork}`);
+      }
+      setReady(true);
+    },
+    [],
+  );
+
+  const signMessageAndLogin = useCallback(
+    async (isLogin: boolean, setErrorMessage: (msg: string) => void) => {
```

Comment:
add return type

Reply:
There is already defined return type here (void) on line 134.